pyomo/pysp/tests/rapper/rapper_tester.py

```diff
@@ -110,6 +111,19 @@ def test_ef_solve_with_gap(self):
                                 tree_model = self.farmer_concrete_tree)
         res, gap = stsolver.solve_ef(solvername, tee=True, need_gap=True)
 
+    @unittest.skipIf(not solver_available,
+                     "%s solver is not available" % (solvername,))
+    def test_ef_solve_with_csvwriter(self):
+        """ solve the ef and report gap"""
+        stsolver = rapper.StochSolver("ReferenceModel.py",
+                                      fsfct = "pysp_instance_creation_callback",
+                                tree_model = self.farmer_concrete_tree)
+        res, gap = stsolver.solve_ef(solvername, tee=True, need_gap=True)
+        csvw.write_csv_soln(stsolver.scenario_tree, "testcref")
+        with open("testcref.csv", 'r') as f:
+            line = f.readline()
+        assert(line.split(",")[0] == "FirstStage")
```

Comment:
This test needs to delete the file it created:
```python
fname = "testcref.csv"
try:
    csvw.write_csv_soln(stsolver.scenario_tree, fname)
    with open(fname, 'r') as f:
        data = f.readline(s)
finally:
    if os.path.exists(fname):
        os.remove(fname)
```
Other notes:
- This should probably use the tempfile creation logic from either python or pyutilib - we cannot always guarantee that the CWD is writable.
- can we test more than the first entry of the first line?  For example, a sanity check that the remainder of the file is not empty?